File: Y2020/01/part1.c
#include <stdio.h>

int main() {
    FILE *fp = fopen("data.txt", "r");
    int nums[1024];
    int count = 0;
    int num;
    while(fscanf(fp, "%d\n", &num) == 1) {
        nums[count++] = num;
    }
    printf("%d\n", count);
    for(int i = 0; i < count - 1; i++) {
        for(int j = i + 1; j < count; j++) {
            if(nums[i] + nums[j] == 2020) {
                printf("%d\n", nums[i] * nums[j]);
                return 0;
            }
        }
    }
    return 0;
}

File: Y2023/D04/part1.js
const fs = require('fs')

const input = fs.readFileSync(`${__dirname}/data.txt`, 'utf8').split('\n')

let total = 0;
for(const card of input) {
  const [_, winNumStr, numStr] = card.match(/Card\W+\d+: ([\d ]+) \| ([\d ]+)/);
  const winNums = winNumStr.replaceAll('  ', ' ').trim().split(' ').map(Number);
  const nums = numStr.replaceAll('  ', ' ').trim().split(' ').map(Number);
  let cardTotal = 0;
  for(const num of nums) {
    if(winNums.includes(num)) {
      if(cardTotal === 0) {
        cardTotal = 1;
      } else {
        cardTotal *= 2;
      }
    }
  }
  total += cardTotal;
}
console.log(total)